create.ps1

```diff
@@ -1,274 +1,378 @@
+                        })
+                }
+                else {
+                    Write-Warning "DryRun: Would create account [$($account.Username)]"
+                    Write-Warning "DryRun: Body: $($createUserSplatParams.Body)"
+                }
+            }
+            catch {
+                $ex = $PSItem
+                if ($($ex.Exception.GetType().FullName -eq "Microsoft.PowerShell.Commands.HttpResponseException") -or
+                    $($ex.Exception.GetType().FullName -eq "System.Net.WebException")) {
+                    $errorObj = Resolve-HelloIDError -ErrorObject $ex
+                    $auditMessage = "Error creating account [$($account.Username)]. Error: $($errorObj.FriendlyMessage)"
+                    Write-Warning "Error at Line [$($errorObj.ScriptLineNumber)]: $($errorObj.Line). Error: $($errorObj.ErrorDetails)"
+                }
+                else {
+                    $auditMessage = "Error creating account [$($account.Username)]. Error: $($ex.Exception.Message)"
+                    Write-Warning "Error at Line [$($ex.InvocationInfo.ScriptLineNumber)]: $($ex.InvocationInfo.Line). Error: $($ex.Exception.Message)"
+                }
+                $outputContext.AuditLogs.Add([PSCustomObject]@{
+                        # Action  = "" # Optional
+                        Message = $auditMessage
+                        IsError = $true
+                    })
+
+                # Throw terminal error
+                throw $auditMessage
+            }
+
+            break
+        }
+
+        "CorrelateAccount" {
+            $outputContext.AccountReference = $correlatedAccount.userGUID
+            $outputContext.Data = $correlatedAccount
+
+            $outputContext.AuditLogs.Add([PSCustomObject]@{
+                    Action  = "CorrelateAccount" # Optionally specify a different action for this audit log
+                    Message = "Correlated to account [$($correlatedAccount.Username)] with AccountReference: $($outputContext.AccountReference | ConvertTo-Json) on field: [$($correlationField)] with value: [$($correlationValue)]"
+                    IsError = $false
+                })
+
+            $outputContext.AccountCorrelated = $true
+
+            break
+        }
+        
+        "MultipleFound" {
+            $auditMessage = "Multiple accounts found where [$($correlationField)] = [$($correlationValue)]. Please correct this so the accounts are unique."
+
+            $outputContext.AuditLogs.Add([PSCustomObject]@{
+                    # Action  = "" # Optional
+                    Message = $auditMessage
+                    IsError = $true
+                })
+        
+            # Throw terminal error
+            throw $auditMessage
 
-Write-Output $result | ConvertTo-Json -Depth 10
\ No newline at end of file
+            break
+        }
+    }
+}
+catch {
+    $ex = $PSItem
+    Write-Warning "Terminal error occurred. Error Message: $($ex.Exception.Message)"
```

Comment:
This comment applies to all scripts where this catch / write-warning is used
The auditlog and this warning are shown in the logging. I guess Write-Verbose is better if debugging is required.

Reply:
This is by design, as you'd want to actually know when a terminal error has occurred.
For isntance the audit log may show several errors, but all non critical/terminal.
Then the write-warning actually shows the terminal/critical error only.
Let's discuss this when possible.